### PR TITLE
Move registering blocks to its own heading

### PR DESCRIPTION
**Pull request by ahuang11**
### Summary

It was hard to find how to add it to Cloud blocks, plus registering is not just for new block types, so I moved it to its own heading.

### Checklist

This pull request is:

- [x] A documentation / typographical error fix
	- No tests or issue needed
- [ ] A short code fix
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a bug report issue 
	- Please include tests. One-line fixes without tests will not be accepted unless it's related to the documentation only.
- [ ] A new feature implementation
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a feature enhancement issue 
	- Please include tests
    - Please make sure that your QA steps are both thorough and easy to reproduce by somebody with limited knowledge of the feature that you are submitting

**Happy engineering!**

### CODE REVIEW

**Comment by ahuang11** [comment]
image

File: docs/concepts/blocks.md [hunk 1]
```diff
@@ -264,3 +246,22 @@ my_s3_bucket.save("my_s3_bucket")
 ```
 
 In the above example, the values for `AWSCredentials` are saved with `my_s3_bucket` and will not be usable with any other blocks.
+
+## Registering blocks to display on Prefect Cloud
```
Comment (desertaxle) [inline docs/concepts/blocks.md:250]:
```suggestion
## Registering blocks for use in Prefect Cloud
```

**Review by serinamarie — APPROVED** [review]
Looks great to me! Thanks for doing this.

**Review by tpdorsey — APPROVED** [review]
Looks good. Thanks

**Review by desertaxle — APPROVED** [review]

**Pull request merged**